### PR TITLE
BUG: linalg.det: Fix edge case handling of stacked 1x1 arrays

### LINKED ISSUE #21582: BUG: Squeezed output from batched `scipy.linalg.det`
### Describe your issue.

According to the documentation, the output of `scipy.linalg.det` for an array shaped like `(l1, l2,..., lx, N, N)` should be `(l1, l2,..., lx)`. However, if N=1 and some of the `li` are also 1, those get squeezed by the scipy implementation. The numpy implementation handles these cases correctly.

### Reproducing Code Example

```python
array = numpy.array([0,1]).reshape(2,1,1,1)
print(numpy.linalg.det(array).shape) # prints (2, 1) as expected
print(scipy.linalg.det(array).shape) # prints (2,)
```

### Error message

```shell
None
```

### SciPy/NumPy/Python version and system information

```shell
()
1.13.0 1.24.2 sys.version_info(major=3, minor=11, micro=9, releaselevel='final', serial=0)
/usr/lib64/python3.11/site-packages/scipy/__config__.py:154: UserWarning: Install `pyyaml` for better output
  warnings.warn("Install `pyyaml` for better output", stacklevel=1)
{
  "Compilers": {
    "c": {
      "name": "gcc",
      "linker": "ld.bfd",
      "version": "10.4.0",
      "commands": "gcc-10",
      "args": "-fmessage-length=0, -grecord-gcc-switches, -O2, -Wall, -D_FORTIFY_SOURCE=2, -fstack-protector-strong, -funwind-tables, -fasynchronous-unwind-tables, -fstack-clash-protection, -g, -fno-strict-aliasing",
      "linker args": "-fmessage-length=0, -grecord-gcc-switches, -O2, -Wall, -D_FORTIFY_SOURCE=2, -fstack-protector-strong, -funwind-tables, -fasynchronous-unwind-tables, -fstack-clash-protection, -g, -fno-strict-aliasing"
    },
    "cython": {
      "name": "cython",
      "linker": "cython",
      "version": "3.0.10",
      "commands": "cython"
    },
    "c++": {
      "name": "gcc",
      "linker": "ld.bfd",
      "version": "10.4.0",
      "commands": "g++-10"
    },
    "fortran": {
      "name": "gcc",
      "linker": "ld.bfd",
      "version": "10.4.0",
      "commands": "gfortran-10"
    },
    "pythran": {
      "version": "0.15.0",
      "include directory": "../../../../../../usr/lib/python3.11/site-packages/pythran"
    }
  },
  "Machine Information": {
    "host": {
      "cpu": "x86_64",
      "family": "x86_64",
      "endian": "little",
      "system": "linux"
    },
    "build": {
      "cpu": "x86_64",
      "family": "x86_64",
      "endian": "little",
      "system": "linux"
    },
    "cross-compiled": false
  },
  "Build Dependencies": {
    "blas": {
      "name": "openblas",
      "found": true,
      "version": "0.3.26",
      "detection method": "pkgconfig",
      "include directory": "/usr/include/openblas",
      "lib directory": "/usr/lib64/openblas-pthreads",
      "openblas configuration": "USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS= NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP=0 ZEN MAX_THREADS=64",
      "pc file directory": "/usr/lib64/pkgconfig"
    },
    "lapack": {
      "name": "openblas",
      "found": true,
      "version": "0.3.26",
      "detection method": "pkgconfig",
      "include directory": "/usr/include/openblas",
      "lib directory": "/usr/lib64/openblas-pthreads",
      "openblas configuration": "USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS= NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP=0 ZEN MAX_THREADS=64",
      "pc file directory": "/usr/lib64/pkgconfig"
    },
    "pybind11": {
      "name": "pybind11",
      "version": "2.12.0",
      "detection method": "pkgconfig",
      "include directory": "/include"
    }
  },
  "Python Information": {
    "path": "/usr/bin/python3.11",
    "version": "3.11"
  }
}
```

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Closes gh-21582.

### CODE REVIEW

**Review by lucascolley — APPROVED** [review]
thanks Warren! LGTM with one question

File: scipy/linalg/tests/test_basic.py [hunk 1]
```diff
@@ -1089,13 +1090,13 @@ def test_1x1_all_singleton_dims(self):
         a = np.array([[[[1]]]], dtype='f')
         deta = det(a)
         assert deta.dtype.char == 'd'
-        assert np.isscalar(deta)
```
Comment (lucascolley) [inline scipy/linalg/tests/test_basic.py:1092]: why remove the scalar check?

Comment (WarrenWeckesser) [reply scipy/linalg/tests/test_basic.py:1092]: Because the result is not a scalar. It is an array with shape (1, 1).

Comment (lucascolley) [reply scipy/linalg/tests/test_basic.py:1092]: ah of course :man_facepalming: makes sense!

**Comment by WarrenWeckesser** [comment]
Thanks @lucascolley, including thanks for--once again--the copy-edit of the title!

**Comment by lucascolley** [comment]
> including thanks for--once again--the copy-edit of the title!

no worries, I do it to everyone :)

**Pull request merged**